### DISCUSSION TITLE
Access audio CD from Webbrowser

### DISCUSSION THREAD

**Discussion by camots** [aldostools/webMAN-MOD#813] [General]
Hi,

I would like to access the files stored on a audio disc from the Webbrowser of another device in order to play it there locally. The PS3 would just serve the .aiff files. How can I do this?

Thank you for your help.

**Comment by aldostools** [comment]
webMAN MOD serves the files on /dev_bdvd as they are recognized by the native file system of PS3. 

The other device simply access these files through http or ftp.

**Reply by camots** [reply]
Thank you for the quick reply.

If I cannot see any dev_bdvd on the client-site i.e. through the browser on the other device.
On the XMB, the disk is mounted and its audio content can be played.
On the webMAN MOD client site (the other device) I don't see dev_bdvd. I do have the dev_hdd0, however I was not able to find the mount-point of the disk in anywhere...

Best regards!

**Comment by aldostools** [comment]
As I commented earlier, webMAN MOD serves what it can access through the native file system. If the CD doesn't show a mount entry, you would have to mount it manually using the /mount.ps3 command and indicating the file system.

I haven't done that before, so you should try yourself.

You can also import the tracks to hdd0 and use them instead of use the CD directly.

**Reply by camots** [reply]
I was thinking that if the CD was showing up in XMP, it would have been mounted by the native file-system of the PS3 and thus, a mount-point would already exist. This seems not to be the case?

I'll try to follow your suggestion to mount the drive manually (becuase I'm new to the CFW world of EvilNat 4.89 cobra 8.3, I first have to lookup for the syntax to do so).

The reason I would like to access the content from the  CD directly: We have a lot of audio CD's we share with conventional CD drives. webMAN MOD would be a great tool for sharing the PS3 drive across the local network i.e. it would turn the PS3 into a network CD (or even DVD?) player.

Thank you for your help!

Best regards

**Comment by aldostools** [comment]
You can try sharing the CD drive using ps3netsrv and mounting the remote drive as /dev_bdvd on the PS3

After that you can access the files through http or ftp from PS3. But this requires to have a PC with DVD drive available.

**Reply by camots** [reply]
My goal is to share the PS3 drive through webMan without involving further resources.
I do the same with the DVD drive of a Mac. It is super easy to do.
Because of webMan, I thought it would be easily possible with the PS3 as well.
Imagine, you could just power on the PS3, insert the CD or DVD, go to an iPad, access the disk through webMan and airPlay it to a device of your choice...

**Comment by aldostools** [comment]
I couldn't find references for CD files in the wiki. Maybe it needs to be mounted as CELL_FS_IOS:PATA0_BDVD_DRIVE or CELL_FS_ISO9660
https://www.psdevwiki.com/ps3/Files_on_the_PS3#Mountpoints

I haven't done before what you want. If someone describe an easy way to access these files, I could implement it in webMAN MOD.

**Reply by camots** [reply]
"I haven't done before what you want. If someone describe an easy way to access these files, I could implement it in webMAN MOD."

This would be great!!!

I will try my best to figure out how to do it.
Coming from unix, I feel it should be possible to find out how PS3 accesses the drive. Somehow it must be already mounted in its native file-system since XMB can access the content. Or is it simply a problem of access rights?
Can the drive be mounted through webMAN MOD if it is already mounted by XMB? In unix, trying to mount a drive which is already mounted results in error.

**Comment by aldostools** [comment]
It isn't due access rights. It's due the CD is accessed by XMB without create a mount point.

It need some tests to find a method to mount the inserted CD to make it visible by the file system... I cannot dedicate time to this task.

Some of the tests involves using sys_fs_mount() as the example in the wiki. 
The web command /mount.ps3 provides a web interface to perform that command, in case you don't have access to psl1ght or SDK.

**Reply by camots** [reply]
I'll investigate this and let you know...

By the way: I'm also running multiMan on the PS3. There is a FTP server as well. What would happen if this would be running as well? Will there be some interference since 2 FTP servers would be running on the PS3?

**Reply by Masamune3210** [reply]
Just a thought, but it might be due to the fact that a Audio CD doesn't actually have a filesystem in the commen sense. The only reason that computers show one is because they make a virtual one for the benefit of the user

**Reply by camots** [reply]
According to the link below Audio CD's have a filesystem.
It is ISO9669 (although many extensions exist)

https://stackoverflow.com/questions/1026909/what-are-the-file-system-of-cd-and-dvd

If I inspect the Audio-CD in macos using diskutil, I get

/dev/disk2 (internal, physical):
   #:                       TYPE NAME                    SIZE       IDENTIFIER
   0:        CD_partition_scheme Audio-CD               *429.5 MB   disk2

**Reply by Masamune3210** [reply]
ISO-9660 is for storing data on a CD, a Audio CD like you would get from the store is not under that, it is under the Red Book standard. There is no filesystem, almost the entire disc is just one long end to end stream of audio data with only a couple other bits of data strewn around, mostly the TOC and some other metadata.

**Reply by Masamune3210** [reply]
![Screenshot_20221212-140925.jpg](https://user-images.githubusercontent.com/1053504/207144999-37f71946-3843-4559-bc7f-db9ae010ab02.jpg)

**Reply by camots** [reply]
I think you are right!
As a test, I have burned the CD in a CD Data format (not audio format).
After inserting the CD, the dev_bdvd could be accessed through webMAN MOD and the files were all there!

Thx for clarification!

**Reply by Masamune3210** [reply]
No problem, you can probably imagine the amount of times I see people getting this confused, and the fact that they show up as files on a computer no matter what doesn't aid in clearing up the confusion much lol. Iirc, the firmware has a prx the XMB calls into to play audio cd's, the contents themselves never make it to the filesystem other than maybe a cache unless you rip the audio first

**Comment by camots** [comment]
After clarification by Masamune3210 (thanks again), I guess the functionality can be achieved only by implementing suport for   CDfs.

https://en.wikipedia.org/wiki/CDfs
